### ISSUE TITLE
Feature: restrict creating of pastes

### ISSUE THREAD

**Issue by jorunkel** [PrivateBin/PrivateBin#719]
Hi there,

I use PrivateBin as a self-hosted installation but on a public server. Since I want to use it only to share data between me and my clients how can I restrict the creation of pastes? I see several ways to do that:

1. Password-protect the whole installation using an .htaccess file. While this is pretty simple to setup it is quiet annoying to access the created pastes (possibly forcing the user to type 2 different passwords).
2. Add a configuration to the PrivateBin so that the creation of pastes is only available with a creation password (shared among the owner and his clients).
3. Require a user login so that users need to authenticate with name + password before creating a paste.

**Comment by outis151** [comment]
I agree, this is a needed feature especially for paste instances that have file upload enabled. The best option would be option number 2 in my opinion.

**Comment by r4sas** [comment]
RTFM: https://github.com/PrivateBin/PrivateBin/wiki/Restrict-upload-using-NGINX

**Comment by jorunkel** [comment]
Option 4: use NGINX as described in the wiki. This is not very straight-forward, though, as NGINX is not available on shared hostings.

**Comment by r4sas** [comment]
Read apache documentation about writing htaccess files in that case.

**Comment by elrido** [comment]
I would not recommend to introduce such a feature into this project. Keep in mind that adding such a feature implies implementing some kind of authentication. Currently this software is fully designed around purely anonymous usage, which simplifies a lot of things and avoids a lot of attack surfaces.

I would go with option 1 (basic access authentication) is the best solution for this use case, at this point. As far as I understand the use case, you are willing to share the access to your instance with the clients, but not with random visitors. After you create the paste, you can stick the [basic auth username / password into the created URL](https://en.wikipedia.org/wiki/Basic_access_authentication#URL_encoding) and send that to the client. They will click the link, the credentials are sent along, but not shown in the browsers address bar, and they'll remain authenticated until they close the browser.

**Comment by jorunkel** [comment]
Thanks for the explanatory answer. The idea of sending username + password stored in the URL has the disadvantage that these credentials are given in plain text and the [browser support is regressing](https://bugs.chromium.org/p/chromium/issues/detail?id=82250#c7).

Anyways, I ended up using the .htaccess file and setup Basic Auth just for everything except GET requests. That way everyone can access the PrivateBin but only those knowning the credentials can actually create new pastes. Here is what I used for this:

```
AuthType Basic
AuthName "Password Protected Area"
AuthUserFile "/path/to/.htpasswd"
<LimitExcept GET>
	Require valid-user
</LimitExcept>
```

Someone might want to add this to the manual.
